docs/proposals/20240808-enhance-operational-efficiency-of-K8s-cluster-in-user's-IDC.md

```diff
@@ -0,0 +1,125 @@
+1. Create host-K8s in cloud service provider, utilizing the abilities of cloud service provider and cloud-edge collaboration of OpenYurt.
+2. Access three machines in user's IDC to host-K8s as worker nodes.
+3. Deploy control plane components of tenant-K8s on worker nodes of host-K8s. All components are deployed in form of `pod` and in `HostNetwork` mode.
+4. Access the remaining machines in user's IDC to tenant-K8s, as worker nodes of tenant-K8s.
+<div align="center">
+<img src="../img/enhance-efficiency-of-K8s-cluster-in-user's-IDC/fig2.png" width=40% />
+</div>
+
+### Architecture
+
+We demonstrate the design of tenant-K8s in Fig.3.
+<div align="center">
+<img src="../img/enhance-efficiency-of-K8s-cluster-in-user's-IDC/fig3.png" width=90% />
+</div>
+
+In tenant-K8s, the designed details are as follows:
+- In control plane nodepool, we will afford users the template of control plane components:
+  1. KCM and Scheduler are deployed as `deployment`, which both have two replicas. KCM and Scheduler access KubeAPIServer by it's service.
+  2. KubeAPIServer is deployed as `daemonset`. KubeAPIServer access ETCD by service of ETCD.
+  3. There are two type of ETCD: data and event, which are both deployed as `statefulset`.
+
+- In local nodepool:
+  1. We add a new `local` mode in YurtHub. In `local` mode, YurtHub will maintain a loadbalance rule, allowing components like Kubelet to load balancing access to the KubeAPIServer.
+  2. yurtadm join affords users to access nodes in their own IDCs to tenant-K8s in `local` mode.
+
+`local` mode YurtHub gets pod's ip in host-K8s's apiserver, and maintains the loadbalance rule to afford load balancing access to APIServer-pods, which is shown in Fig.4.
+<div align="center">
+<img src="../img/enhance-efficiency-of-K8s-cluster-in-user's-IDC/fig4.png" width=90% />
+</div>
+
+### User stories
+
+#### Comparison
+We compare above mentioned solutions and proposed scheme, shown in the table:
+|  | solution1 | solution2 | solution3 | proposed |
+| --- | --- | --- | --- | --- |
+| architecture | <img src="../img/enhance-efficiency-of-K8s-cluster-in-user's-IDC/solution1.png" width="273"> | <img src="../img/enhance-efficiency-of-K8s-cluster-in-user's-IDC/solution2.png" width="274"> | <img src="../img/enhance-efficiency-of-K8s-cluster-in-user's-IDC/solution3.png" width="268"> | <img src="../img/enhance-efficiency-of-K8s-cluster-in-user's-IDC/fig2.png" width="303"> |
+| operational efficiency | poor | moderate | good | good |
+| security | poor | poor | good | good |
+| support multi-tenant | poor | moderate | poor | moderate |
+| Deployment simplicity | poor | poor | good | moderate |
```

Comment:
Deployment simplicity --> Installation Difficulty

Reply:
Thanks for comments, done.